consensus/ethash/consensus.go

```diff
@@ -204,11 +204,20 @@ func (ethash *Ethash) VerifyUncles(chain consensus.ChainReader, block *types.Blo
 
 	number, parent := block.NumberU64()-1, block.ParentHash()
 	for i := 0; i < 7; i++ {
+		ancestorHeader := chain.GetHeader(parent, number)
+		if ancestorHeader == nil {
```

Comment:
Just one concern, if we lose the header because the database reason, we may loosen rules for checking block.

Reply:
Yes, I know. It might be worth having a `panic` here, but I didn't want to introduce one now -- since the current code just silently does the exact same thing